### PR TITLE
Implement deposit/withdraw SDK

### PR DESCRIPTION
**Pull request by vcarl**
This still needs an example, and I haven't actually tested it at all, but all the types and logic should be in place.

### CODE REVIEW

File: src/sep6/DepositProvider.ts [hunk 1]
```diff
@@ -0,0 +1,65 @@
+import queryString from "query-string";
```
Comment (morleyzhi) [inline src/sep6/DepositProvider.ts]: I think we should not call this "sep6", it's too implementation-specific. I would call it Transfers or DepositWithdraw or something. The former is probably cleanest.

File: src/sep6/getKycUrl.ts [hunk 2]
```diff
@@ -0,0 +1,6 @@
+import { InteractiveKycNeeded } from "./types";
+
+export const getKycUrl = (
```
Comment (morleyzhi) [inline src/sep6/getKycUrl.ts]: Should probably be a named function.

File: src/sep6/parseInfo.ts [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+  Fee,
+  Field,
+  RawField,
+  RawInfo,
+  RawType,
+  SimpleFee,
+  WithdrawInfo,
+} from "./types";
+
+export const parseInfo = (info: RawInfo) => {
```
Comment (morleyzhi) [inline src/sep6/parseInfo.ts]: Can be a named function.

File: src/util.d.ts [hunk 4]
```diff
@@ -0,0 +1 @@
+export type Omit<T, K extends keyof T> = Pick<T, Exclude<keyof T, K>>;
```
Comment (morleyzhi) [inline src/util.d.ts]: I think this type can be named to better describe what it is.

File: src/sep6/types.d.ts [hunk 5]
```diff
@@ -0,0 +1,205 @@
...
+}
+
+export interface FeeArgs {
+  assetCode: string;
+  amount: string;
+  operation: "withdraw" | "deposit";
+  type: string;
+}
+
+export interface RawInfo {
```
Comment (morleyzhi) [inline src/sep6/types.d.ts]: I think this type name could be more descriptive, this seems like "TransferInfoResponse".

File: src/sep6/DepositProvider.ts [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+
+import { Omit } from "../util";
+import { fetchKycInBrowser } from "./fetchKycInBrowser";
+import { TransferProvider } from "./TransferProvider";
+import {
+  DepositRequest,
+  FeeArgs,
+  InteractiveKycNeeded,
+  TransferResponse,
+} from "./types";
```
Comment (morleyzhi) [inline src/sep6/DepositProvider.ts]: In my last revision, I had the types inherit from `src/types/index.ts` which imports `src/types/data.ts` and `src/types/keys.ts`. Maybe those should be .d.ts files instead, but either way, we should have all three arms be consistent with each other.

File: src/sep6/DepositProvider.ts [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+  }
+
+  public async fetchKycInBrowser({
+    response,
+    request,
+    window: windowContext,
+  }: {
+    response: InteractiveKycNeeded;
+    request: DepositRequest;
+    window: Window;
```
Comment (morleyzhi) [inline src/sep6/DepositProvider.ts]: You may want to check what type React Native's webview packages return?

https://facebook.github.io/react-native/docs/webview

File: src/sep6/WithdrawProvider.ts [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+      operation: "withdraw",
+    });
+  }
+
+  public async fetchKycInBrowser({
+    response,
+    request,
+    window: windowContext,
+  }: {
+    response: InteractiveKycNeeded;
```
Comment (morleyzhi) [inline src/sep6/WithdrawProvider.ts]: Looking at this without knowing intimately how the system works, I wouldn't know what shape "response" was supposed to be. I would have guessed `InteractiveKycNeeded` was a boolean. So maybe the name can give clearer hints to what this actually is?

File: src/sep6/DepositProvider.ts (hunk 1 above)
Comment (vcarl) [reply src/sep6/DepositProvider.ts]: Ah whoops! Renamed it everywhere but the folder.

File: src/sep6/types.d.ts (hunk 5 above)
Comment (vcarl) [reply src/sep6/types.d.ts]: I like `Raw` to signify that it's unprocessed, but yeah I'll make it a little more verbose.

File: src/sep6/DepositProvider.ts (hunk 7 above)
Comment (vcarl) [reply src/sep6/DepositProvider.ts]: I looked into that briefly, react-native appears to use a non-standard PostMessage that I'm not sure we should specifically target. I'm thinking we leave it  for now and circle back once we can dogfood/get feedback from a react-native use case.

Though I do see that I forgot to add the original request to the callback URL like we'd talked about, I'll add that.

**Pull request merged**